### PR TITLE
fix `DartType.isDynamic` deprecation 

### PR DESCRIPTION
**Pull request by SunlightBro**
With analyzer 5.12.0 
`DartType.isDynamic` got deprecated,
instead use `this is DynamicType`.

### CODE REVIEW

**Comment by rrousselGit** [comment]
That's a wild change IMO, but sure. Will review a bit later :)

**Comment by rrousselGit** [comment]
Yeah I was reading that.

I would assume that yes it is impacted. But I'll have to double check when I'll give this a proper review

**Comment by SunlightBro** [comment]
You can revert [`8490802`](https://github.com/rrousselGit/freezed/pull/903/commits/84908024240e74efcaded0602c90744537340b99) if it's not something you want.
I just wanted to see if it would run the Checks successfully.

**Comment by rrousselGit** [comment]
Thanks for fixing the CI~

File: packages/freezed/lib/src/freezed_generator.dart [hunk 1]
```diff
@@ -167,7 +167,7 @@ Read here: https://github.com/rrousselGit/freezed/blob/master/packages/freezed/C
       if (parameter.type.nullabilitySuffix != NullabilitySuffix.question &&
           parameter.isOptional &&
           parameter.defaultValue == null &&
-          !parameter.type.isDynamic) {
+          !(parameter.type is DynamicType)) {
```
Comment (rrousselGit) [inline packages/freezed/lib/src/freezed_generator.dart:170]:
```suggestion
          parameter.type is! DynamicType) {
```

**Comment by rrousselGit** [comment]
I think InvalidType should be checked on all of these cases too.

One of the issues is what could happen when the output of a code generator is the input of a Freezed class.  
We'd need to test something that for all of these `is DynamicType` cases

For example for the "is possibly a collection", someone could do:

```dart
@freezed
class Example with _$Example {
  factory Example(
    GeneratedClassWhichImplementsList list,
  );
}
```

where `GeneratedClassWhichImplementsList` is not available at the moment where Freezed runs.

**Comment by rrousselGit** [comment]
Before there was no distinction between dynamic and invalid, so the tests simply used dynamic to mimic an invalid type. But that's not the case anymore so we'd probably have to duplicate the tests for those.

**Pull request merged**